ADDED THE DARK AND LIGHT MODE FUNCTIONALITY USING CSS ONLY ISSUE (#108)

Chirag77302 opened the pull request:
Hey everyone , 
I have added the dark and light mode button functionality which changes the theme of the website to night/dark mode on clicking and reverts it back to the original theme on clicking the light mode , as mentioned in issue (#108) 
Do give out your suggestions for the changes made .

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/twitter/opensource-website?pullRequest=123) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/twitter/opensource-website?pullRequest=123) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/twitter/opensource-website?pullRequest=123) it.</sub>

Chirag77302 commented:
Thankyou @willnorris for your review , the changes were made in response to issue (#108 ) , wherein I felt that adding a standard and decent dark - light feature functionality would provide as a simple and decent additional feature to the website , and may also lead to further developments for the same in future by the community , similar to the one in the standard twitter website .

willnorris commented:
Yeah, the fact that the main Twitter app supports dark mode is one big thing that convinces me that it's fine to add here.  But since we don't already have any "settings" for the site, and the fact that the current implementation doesn't persist the preference, I think the prefers-color-scheme approach is probably the best, at least for the initial implementation.  Then it can kind of serve as a hidden easter egg for folks that have that enabled in their browser.  

What do you think?  Would you be up for switching this PR to use prefers-color-scheme?

Chirag77302 commented:
Yeah , thankyou @willnorris , I am up for this task .

Chirag77302 commented:
I totally agree with you @willnorris , still I think we can merge the given changes just for additional developments to the site , in the meantime I will add the necessary 'prefer-color-scheme' changes to it .

Chirag77302 commented:
All the necessary changes have been made @willnorris , talking of the javascript code , it doesn't seem to work in the `/static/js/main.js` file or the `/static/js/index.js` file , therefore I just cleaned up the code for the same function as you wanted it to be , and added the function to the` footer.html` file , just so that it does not get fumbled up with the google analytics .

Also the classes regarding toggle-button were removed earlier only , only the necessary ones which were needed to maintain the **`colour scheme pattern`** for the  **'_OS PAGE OF THE WEBSITE_'** have been kept . Rest all have been cleaned up .

Chirag77302 commented on assets/css/_feather-component-button.scss:
```diff
@@ -306,12 +305,24 @@ fieldset[disabled] a.Button {
-  background-color: #e8f6fe;
-  border-color: #1DA1F2;
-  color: #1DA1F2;
+
```
This part of code was added in the original file itself , before any commits were made by me , therefore I preferred to leave it as it was .

willnorris commented on assets/css/projects.scss:
```diff
@@ -82,7 +82,7 @@
 
 .search-box::placeholder {
-	color: white;
+	color: black;
```
as noted elsewhere, don't change the default styling.  Only add overrides for dark mode

willnorris commented on assets/css/_main.scss:
```diff
@@ -354,3 +361,21 @@ br {
 }
 
+
+.explore-projects{
```
you shouldn't need to define any additional styling for the default (light) theme.  The only CSS additions should be dark mode overrides in the right locations.

For example, for what you're trying to do here (as well as with .section and .all-projects below) is better done simply by overriding the background and color on the body class up around line 45:

``` css
body {
  background-color: white;

  @media (prefers-color-scheme: dark) {
    background-color: var(--tw-color-gray-900-experimental);
    color: var(--tw-color-gray-faint);
  }
}
```

willnorris commented on assets/css/projects.scss:
```diff
@@ -37,7 +37,7 @@
 }
 
-.project-links a:link, .project-links a:visited {
+a:link, .project-links a:visited {
```
this shouldn't be changed.

willnorris reviewed (commented):
> talking of the javascript code , it doesn't seem to work in the /static/js/main.js file or the /static/js/index.js file

It definitely does work in both main.js or index.js (though index.js is probably better, since this is only needed on the homepage).  You might need to restart Hugo or force refresh in your browser?

> Also the classes regarding toggle-button were removed earlier only , only the necessary ones which were needed to maintain the colour scheme pattern for the 'OS PAGE OF THE WEBSITE' have been kept . Rest all have been cleaned up .

I'm not really sure what you mean by this.  There are still a lot of inline styles and transitions directly in the html files, as well as some spurious whitespace changes.  You shouldn't need to edit any of the html files.

I took a pass at these changes to verify what I was saying was true.  If you want to take a peek, you can [see the changes here](https://github.com/willnorris/opensource-website/commit/70be8a8ef55ca8493eeddbbfe5b75781e18c9088) (ignore the Button styles... I'm still not sure of the best way to handle that.)

willnorris commented:
for simplicity, I went head and reduced your changes down to the core CSS changes and merged them as 76981e0da5d65c6ca04ce6fe74d97473b189942d.  I then did a little bit more refactoring and took a first stab at customizing the buttons.  This should give us a good foundation now to continue to iterate on tweaking dark mode.  Thanks for your work and patience on this!

Chirag77302 commented:
> for simplicity, I went head and reduced your changes down to the core CSS changes and merged them as [76981e0](https://github.com/twitter/opensource-website/commit/76981e0da5d65c6ca04ce6fe74d97473b189942d). I then did a little bit more refactoring and took a first stab at customizing the buttons. This should give us a good foundation now to continue to iterate on tweaking dark mode. Thanks for your work and patience on this!

@willnorris kindly can you please make a confirmation whether the changes have been merged by adding the **'MERGED'** tag to this PR .

willnorris commented:
yes, they have been merged in commit 76981e0da5d65c6ca04ce6fe74d97473b189942d.  It's the same code you provided in this PR and lists you as the author, I just squashed the commits into a single one, and removed some of the unrelated bits.

Unfortunately, because I had to make local edits to the commit like that, GitHub lists this PR as "closed" rather than "merged", and there's no way to change this.  It's a little annoying, but rest assured your changes were actually merged.

Chirag77302 commented:
> yes, they have been merged in commit [76981e0](https://github.com/twitter/opensource-website/commit/76981e0da5d65c6ca04ce6fe74d97473b189942d). It's the same code you provided in this PR and lists you as the author, I just squashed the commits into a single one, and removed some of the unrelated bits.
> 
> Unfortunately, because I had to make local edits to the commit like that, GitHub lists this PR as "closed" rather than "merged", and there's no way to change this. It's a little annoying, but rest assured your changes were actually merged.

Thankyou @willnorris , it was really nice working with you , please feel free to put up any new issues on any of the repositories , I will be happy to contribute . 
P.S - Apologies as I didn't realize the fact that you didn't want me to override the set CSS features for the light mode , I just felt that it my changes improved the UI to some extent . 😅😅 Have a nice day !

willnorris commented:
Ah, interesting... I didn't realize you were intentionally trying to improve the light mode UI as well.  My apologies!

I probably still would have asked you to move that to a separate PR so that I could review that independent of the dark mode changes.  And if you do still have some ideas on improving the UI, by all means, feel free to open a new PR with those changes, as I'd be happy to take a look.

Chirag77302 commented:
> Ah, interesting... I didn't realize you were intentionally trying to improve the light mode UI as well. My apologies!
> 
> I probably still would have asked you to move that to a separate PR so that I could review that independent of the dark mode changes. And if you do still have some ideas on improving the UI, by all means, feel free to open a new PR with those changes, as I'd be happy to take a look.

Sure thing , looking forward to it .
